enterprise/reporting/reporting.go

```diff
@@ -135,7 +138,10 @@ func (r *DefaultReporter) DatabaseSyncer(c types.SyncerConfig) types.ReportingSy
 		MigrationsTable:            "reports_migrations",
 		ShouldForceSetLowerVersion: config.GetBool("SQLMigrator.forceSetLowerVersion", true),
 	}
-	err = m.Migrate("reports")
+	templateData := map[string]int{
+		"AutoVacuumCostLimit": r.autoVacuumCostLimit,
+	}
+	err = m.MigrateFromTemplates("reports", templateData)
```

Comment:
Any reason not applying everytime when the server starts? Also, we would not be requried to ALTER if the configuration is not set
```suggestion
	err = m.Migrate("reports")
    if err != nil {
		panic(fmt.Errorf("could not run reports migrations: %w", err))
	}
	if config.IsSet("Reporting.autoVacuumCostLimit") {
		_, err = dbHandle.ExecContext(ctx, fmt.Sprintf("ALTER TABLE reports SET (autovacuum_vacuum_cost_limit = %d", config.GetInt("Reporting.autoVacuumCostLimit", 200)))
		if err != nil {
			panic(err)
		}
	}
```

Reply:
Done